assert/error.go

```diff
@@ -19,3 +22,13 @@ func Error(t *testing.T, err error, details ...interface{}) {
 		t.Fatalf("expected error, got nil.%s", errordetails(details...))
 	}
 }
+
+// IsError will call Fatal if the given error does not match the want error.
+// It uses the errors.Is() function to check if the error wraps the wanted error.
+func IsError(t *testing.T, want, got error, details ...interface{}) {
```

Comment:
Since errors.Is is also used having "got" before what you want, maybe invert here ?

```suggestion
func IsError(t *testing.T, got, want error, details ...interface{}) {
```

On error reporting on tests I think it is more usual on Go for what you "got"/"actual" to come before what you "want"/"expect":

https://github.com/golang/go/wiki/CodeReviewComments#useful-test-failures

Not related to function signatures, but it seems nice to be consistent about this ordering.